File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> ans;

    void solve(TreeNode* root) {
        if(root != NULL) {
            solve(root->left);
            solve(root->right);
            ans.push_back(root->val);
        }
    }

    vector<int> postorderTraversal(TreeNode* root) {
        solve(root);
        return ans;
    }
};

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def helper(self, root, res):
        if root is not None:
            self.helper(root.left, res)
            self.helper(root.right, res)
            res.append(root.val)
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        self.helper(root, res)
        return res
        